### PR TITLE
Fix parsing compilation database

### PR DESCRIPTION
**Pull request by nowrep**
* fix detecting filename in arguments when filename is relative
 * fix passing absolute paths (eg. for -isystem /usr/include/...)

### CODE REVIEW

**Comment by Shougo** [comment]
Why this fix is needed?
Please explain to me with the example.

**Comment by nowrep** [comment]
Sure:

```json
{
        "command": "c++ -c -pipe -pthread -isystem /usr/include/qt/QtPrintSupport -o ../build/main.o main.cpp",
        "directory": "/home/david/src",
        "file": "/home/david/src/main.cpp"
}
```

In this case, without this patch, the `main.cpp` would not be recognized as source file and passed as argument, which leads to error. Also in this case it would skip the `/usr/include/qt/QtPrintSupport` include.

This case is fixed with not trying to construct full_path when argument is an absolute filepath + testing `os.path.basename` on `used_fname`.

```json
{
        "command": "c++ -c -pipe -pthread -isystem /usr/include/qt/QtPrintSupport -o ../build/main.o dir/main.cpp",
        "directory": "/home/david/src",
        "file": "/home/david/src/dir/main.cpp"
}
```

This case is fixed with testing `full_path == used_fname`.

**Comment by Shougo** [comment]
OK.  Good.

@zchee Ping.

File: rplugin/python3/deoplete/sources/deoplete_clang.py [hunk 1]
```diff
@@ -263,9 +263,11 @@ def get_compilation_database(self, fname):
                 cwd = cmds.directory
                 skip = 1
                 for arg in cmds.arguments:
-                    if skip or arg in \
+                    full_path = os.path.realpath(os.path.join(cwd, arg)) if not os.path.isabs(arg[0])  else ''
```
Comment (Shougo) [inline rplugin/python3/deoplete/sources/deoplete_clang.py]: This line is too long.

**Comment by zchee** [comment]
@nowrep Thanks for the pull request.
As an aside, Now I developing `compile_commands.json` parser written in Go, I had the same problem.

LGTM except shougo suggested point. Could you separate L266 to max of 99 column?

@Shougo Thanks for code review.

**Comment by Shougo** [comment]
Merged.

**Pull request merged**